### DISCUSSION TITLE
Where is Quiche Documentation

### DISCUSSION THREAD

**Discussion by ghost** [netty/netty#11827] [Q&A]
Netty Quic is based on Google version of Quiche (?)
Where can I find the documentation used by the Netty Quic developers in building the Netty Quic Codec.
I am having some difficulty understanding the way Netty Quic Codec builds on Google Quiche.
Google search does not (for me) turn up any links to any helpful docs.

One of my main problems is understanding how QuicChannel is handled by the client and the server.
Does the server maintain persistent record of 0-RTT info?

**Comment by chrisvest** [comment]
Quiche is a QUIC implementation by Cloudflare, primarily: https://github.com/cloudflare/quiche
It implements the RFC versions of QUIC, not the original Google QUIC.
Their repository has further links to their documentation.

**Reply by ghost** [reply]
Thanks, Chris,
Reading through the rust examples gives some insite into the process.
What I am still missing is the way to use Netty QuicChannel over time.
If I create a QuicChannel to an IPAddress, say 23.45.67.55:45678, send some data and finish the streams (FIN frame).
Do I close the QuicChannel or leave it open in case I need it again?
If in 5 mins I want to send some more data to same address, do I use the still open QuicChannel or create a new one?
If the QuicChannels are not closed is there a time after which close them?
How long does the server keep 0-RTT info (In a hashmap it seems from rust code)?
